syftlib/src/main/java/org/openmined/syft/networking/clients/SyftWebSocket.kt

```diff
@@ -108,13 +117,22 @@ internal class SyftWebSocket(
         }
 
          /**
-          *  Handle socket failure and notify subscribers. And try to reconnect
+          *  Handle socket failure and notify subscribers.
+          *  And try to reconnect with exponential backoff delay
           */
         override fun onFailure(webSocket: WebSocket, t: Throwable, response: Response?) {
             super.onFailure(webSocket, t, response)
             statusPublishProcessor.offer(NetworkMessage.SocketError(t))
-            // TODO we probably need here some backoff strategy
-            connect()
+
+            if (++retryCount <= retryPolicy.maxRetryCount) {
+                Handler().postDelayed({
```

Comment:
I think using coroutines is better than using handler here. They are more intuitive and more scalable.